### ISSUE TITLE
Strange issue with guicursor and CPU usage

### ISSUE THREAD

**Issue by xorander00** [tim-harding/neophyte#19]
Recently found this project and I like it! My daily driver is FreeBSD and so I'm slowly working on porting Neovide over to it, but Skia is a major PITA. This compiled and worked fine first time around! :)  
  
So there are a few conditions which seem to cause Neophyte to spike CPU to 100%...  
- When guicursor has blinking disabled (no blink[on|off|wait] specified). When it's blink[on|off|wait]200, it's fine.
- When neophyte is not visible after I switch to another workspace, then it spikes CPU to 100%. This is regardless of whatever guicursor is set to, blinking or not.
  
Haven't had a chance to dig in yet, but figured I'd post here for any thoughts/tips/etc to diagnose this behavior.  
  
Granted, my environment is a bit more esoteric, but the process spiking to 100% CPU is strange...
- OS: FreeBSD 14.1-STABLE (1401501)
- Compositor: Niri (built on Smithay), Wayland
- GPU: AMD Radeon RX 6600 XT  
  
Any thoughts would be appreciated, thanks! :)

**Comment by tim-harding** [comment]
Thanks for investigating this. The note about `guicursor` is very helpful. I'll try and jump in to look at this issue soon, as I've had problems with Neophyte sapping battery life on my Mac and this might be part of the problem. To clarify, are you using the `crates.io` version or the latest from GitHub? I've made updates since the last release that addressed the most egregious runaway resource use, but there may be other problems I haven't found.

**Comment by tim-harding** [comment]
Ah, sorry about that! I unfortunately can't reproduce the issue you're seeing. I add some logging for animation and elapsed time of redraw events, as that seems like the most likely source of problems. Could you run from the latest commit with `RUSTLOG="neophyte=debug" cargo run` and post the output while reproducing the issue?

**Comment by xorander00** [comment]
My bad, my previous comments were from another account. Deleted them. Will try with the latest commit here shortly and let you know.

**Comment by xorander00** [comment]
Issue seems fixed as of 8a4fceb32c7b2923de29cf13ba5586ebd8197229. I can post the logs if you'd still like to see, but from my quick skim, everything seemed pretty normal.  
  
Yesterday I ran it through truss to see what the syscalls looked like when the issue was happening and I saw that during the time when I wasn't on the workspace that had neophyte (i.e. window wasn't visible), the truss log during that time period was ONLY a flood of calls to kevent (part of kqueue on FreeBSD). Seemed like it was trying to unregister an event notification but got stuck in a loop (e.g. iterating over some list but had termination condition bug, or maybe thinking unregistering failed and kept retrying, etc).  
  
I just ran it with truss again and the flood of kevent calls disappeared. Everything looked pretty normal in the log.  
  
Anyway, seems to work now, so whatever changes you made fixed it. Thanks a bunch! :) Will see if I can contribute anything else down the line.
